Rubberduck.Core/UI/Command/Refactorings/RefactorExtractInterfaceCommand.cs

```diff
@@ -32,41 +32,7 @@ private bool SpecializedEvaluateCanExecute(object parameter)
             {
                 return false;
             }
-
-            var interfaceClass = _state.AllUserDeclarations.SingleOrDefault(item =>
-                item.QualifiedName.QualifiedModuleName.Equals(activeSelection.Value.QualifiedName)
-                && ModuleTypes.Contains(item.DeclarationType));
-
-            if (interfaceClass == null)
-            {
-                return false;
-            }
-
-            // interface class must have members to be implementable
-            var hasMembers = _state.AllUserDeclarations.Any(item =>
-                item.DeclarationType.HasFlag(DeclarationType.Member)
-                && item.ParentDeclaration != null
-                && item.ParentDeclaration.Equals(interfaceClass));
-
-            if (!hasMembers)
-            {
-                return false;
-            }
-
-            var parseTree = _state.GetParseTree(interfaceClass.QualifiedName.QualifiedModuleName);
-            var context = ((ParserRuleContext)parseTree).GetDescendents<VBAParser.ImplementsStmtContext>();
-
-            // true if active code pane is for a class/document/form module
-            return !context.Any()
-                   && !_state.IsNewOrModified(interfaceClass.QualifiedModuleName)
-                   && !_state.IsNewOrModified(activeSelection.Value.QualifiedName);
+            return ((ExtractInterfaceRefactoring)Refactoring).CanExecute(_state, activeSelection.Value.QualifiedName);
```

Comment:
Explicitly casting doesn't feel right here. 

I'd rather that it was on an interface or at least an abstract method on the `InteractiveRefactorBase`. This would require modifying other refactoring. If we don't want to address this in this PR, I would at least ask that a `TODO` be added to mark that this needs further development so we don't forget about this tech-debt.